agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -157,8 +157,9 @@ def ensure_terminated
     def service_container_outdated?(service_container, service_pod)
       creator = Kontena::ServicePods::Creator.new(service_pod)
       creator.container_outdated?(service_container) ||
-        creator.labels_outdated?(service_pod.labels, service_container) ||
-          creator.recreate_service_container?(service_container)
+          creator.labels_outdated?(service_pod.labels, service_container) ||
+          creator.recreate_service_container?(service_container) ||
+          creator.image_outdated?(service_pod.image_name, service_container)
```

Comment:
Compared to `Kontena::ServicePods::Creator#service_uptodate?`, this is still missing the `service_container.config['Image'] != service_pod.image_name`. Would that also be needed here?

I'm guessing it's about something like detecting changes to `config['Image']` that happen when new images are pulled and the image tags update to point to new images?